### DISCUSSION TITLE
Spotify Refresh Token not Possible with PKCE

### DISCUSSION THREAD

**Discussion by the-devbear** [supabase/supabase#20035] [Questions]
I was under the [impression](https://supabase.com/docs/reference/javascript/auth-signinwithoauth) that supabase uses PKCE for sign in with providers. Is this not support for Spotify?
If it is supported, then the refreshing for the access_token doesn't work with PCKE. I followed the [guide](https://developer.spotify.com/documentation/web-api/tutorials/refreshing-tokens) from Spotify but I get a 401 Error Code, with invalid request error code.

I'm not sure if this is a question or maybe a bug. But the main question for now would be does supabase does use PCKE for Spotiy

**Comment by J0** [comment]
Supabase uses the PKCE Flow when interacting between Application -> Supabase Auth. However,  between Supabase Auth -> Spotify we currently use the Auth Code Flow

**Reply by joostwmd** [reply]
Is there any plan to move to the PKCE flow between Supabase Auth and e.g Spotify? I am developing native apps and this constraint is really annoying, because there is no good way to refresh the provider token from the native app. I could think of refreshing the token with an edge function, but this seems like a really weird solution to me. Also the point of using supabase is (for me at least) that i don't have to worry about backend stuff...

Is there any particular reason that the Auth Code is used instead of PKCE?

**Reply by henrynguyen7** [reply]
+1 on the above as this effectively makes it so that any SPA or mobile app using Supabase Auth to connect with Spotify can never refresh the token, since such applications cannot safely include the client_secret which would be required to refresh the token without the PKCE flow (more detailed comment here: https://github.com/supabase/auth/issues/1450#issuecomment-2287718083). 

Agree that this reduces the usefulness of supabase as a backend solution, which is unfortunate because everything else provided by Supabase has been fantastic so far 😅
